Sources/PointFree/Account/Cancel.swift

```diff
@@ -0,0 +1,192 @@
+
+    return { conn in
+      let (currentUser, rest) = (conn.data.first, conn.data.second)
+      let subscription = currentUser.subscriptionId
+        .map {
+          AppEnvironment.current.database.fetchSubscriptionById($0)
+            .mapExcept(requireSome)
+            .run
+            .map(^\.right)
+        }
+        ?? pure(nil)
+
+      return subscription.flatMap { sub in
+        guard let sub = sub else {
+          return conn
+            |> writeStatus(.notFound)
+            >-> respond(text: "Not subscribed :(")
+        }
+        return conn.map(const(.init(first: sub, second: .init(first: currentUser, second: rest))))
+          |> middleware
+      }
+    }
+}
+
+func requireSubscriptionOwner<A>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, T3<Database.Subscription, Database.User, A>, Data>
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, T3<Database.Subscription, Database.User, A>, Data> {
+
+    return { conn in
+      guard conn.data.first.userId == conn.data.second.first.id else {
+        return conn
+          |> writeStatus(.notFound)
+          >-> respond(text: "Not the subscription owner :(")
+      }
+
+      return conn |> middleware
+    }
+}
+
+func requireStripeSubscription<A>(_ suchThat: @escaping (Stripe.Subscription) -> Bool)
+  -> (@escaping Middleware<StatusLineOpen, ResponseEnded, T3<Stripe.Subscription, Database.User, A>, Data>)
+  -> Middleware<StatusLineOpen, ResponseEnded, T3<Database.Subscription, Database.User, A>, Data> {
+
+    return { middleware in
+      { conn in
+
+        let (databaseSub, user, rest) = (conn.data.first, conn.data.second.first, conn.data.second.second)
+
+        return AppEnvironment.current.stripe.fetchSubscription(databaseSub.stripeSubscriptionId)
+          .run
+          .map(^\.right)
+          .flatMap { sub in
+
+            guard let sub = sub, suchThat(sub) else {
+              return conn
+                |> writeStatus(.notFound)
+                >-> respond(text: "Stripe subscription not found :(")
+            }
+
+            return conn.map(const(.init(first: sub, second: .init(first: user, second: rest))))
+              |> middleware
+        }
+      }
+    }
+}
+
+let confirmCancelView = View<(Stripe.Subscription, Database.User)> { subscription, currentUser in
+  document([
+    html([
+      head([
+        style(renderedNormalizeCss),
+        style(styleguide),
+        style(render(config: pretty, css: pricingExtraStyles)),
+        meta(viewport: .width(.deviceWidth), .initialScale(1)),
+        ]),
+      body(
+        darkNavView.view((currentUser, nil))
+          <> [
+            gridRow([
+              gridColumn(sizes: [.mobile: 12, .desktop: 8], [style(margin(leftRight: .auto))], [
+                div(
+                  [`class`([Class.padding([.mobile: [.all: 3], .desktop: [.all: 4]])])],
+                  titleRowView.view(unit)
+                    <> formRowView.view(unit)
```

Comment:
This is a pretty basic screen right now (just a title and a big ol button). Gonna add subscription details here, copy around what unsubscribing means, and maybe a set of radio buttons for feedback.